dirwatcher.py

```diff
@@ -0,0 +1,204 @@
+#!/usr/bin/env python
+
+import signal
+import os
+import argparse
+import sys
+import logging
+import time
+import datetime
+
+__author__ = 'tamoyahopkins'
+
+# logging config
+logger = logging.getLogger(__name__)
+logger.setLevel(logging.DEBUG)
+formatter = logging.Formatter('%(asctime)s: %(levelname)s : %(message)s')
+file_handler = logging.FileHandler('dirwatcher.log', 'a')
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+# add stream to include Traceback in logs
+stream_handler = logging.StreamHandler()
+stream_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+# exit option
+close_program = False
+
+
+def action_signal(sig_num, frame):
+    '''Actions incoming signals and closes program if needed'''
+    logger.warning(f'INCOMING SIGNAL - {signal.Signals(sig_num).name}')
+    global close_program
+    close_program = True
+
+
+# handle interval
+interval = 0
+
+# program globals
+current_dict = {}
+state = {}
+
+
+def watch_dir(dir_, search_text, ext=None):
+    '''Loops thru filenames in a dir & searches for magic str.
+        If filename already in dic, updates lineno, if not in dict, gives
+        default value of 0. Returns dict
+    '''
+    # global current_dict
+    if os.path.isdir(dir_):
+        dir_files = []
+
+        for dirpath, _, filenames in os.walk(dir_):
+            for f in filenames:
+                dir_files.append(os.path.abspath(os.path.join(dirpath, f)))
+        if not ext:
+            # if ext is NOT set
+            for file_ in dir_files:
+                # if not in current_dict, make key and set to 0
+                if file_ not in current_dict:
+                    current_dict[file_] = 0
+
+        elif ext:
```

Comment:
How about only add the file to the watch dict if it matches the desired extension?